spark2-sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala

```diff
@@ -133,13 +133,13 @@ object GpuHashJoin {
 
   /** Determine if this type of join supports using the right side of the join as the build side. */
   def canBuildRight(joinType: JoinType): Boolean = joinType match {
-    case _: InnerLike | LeftOuter | LeftSemi | LeftAnti | _: ExistenceJoin => true
+    case _: InnerLike | LeftOuter | LeftSemi | LeftAnti | FullOuter | _: ExistenceJoin => true
```

Comment:
This is confusing. It was confusing before to have `FullOuter` have a left build. Should we have a follow on to pick the best side instead? Also could we add some docs about why this is now build right?

Reply:
In this initial version, I changed this to match LeftOuter join because I was implementing it as a Left Outer join, so I figured I should use the same constraints.  Based on these comments, I agree full join needs to be able to build left of right.  I'm trying to decide if that means I need to use a Right Outer join when building left and Left Outer join when building right?

Reply:
The code now allows build left or build right.